pkg/tfshim/sdk-v2/instance_state.go

```diff
@@ -1,23 +1,46 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package sdkv2
 
 import (
 	"strings"
 
+	"github.com/hashicorp/go-cty/cty"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+
 	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
 	diff_reader "github.com/pulumi/terraform-diff-reader/sdk-v2"
 )
 
 var _ = shim.InstanceState(v2InstanceState{})
 
 type v2InstanceState struct {
-	tf   *terraform.InstanceState
-	diff *terraform.InstanceDiff
+	resource *schema.Resource
+	tf       *terraform.InstanceState
+	diff     *terraform.InstanceDiff
 }
 
-func NewInstanceState(s *terraform.InstanceState) shim.InstanceState {
-	return v2InstanceState{s, nil}
+func NewInstanceState(resource *schema.Resource, s *terraform.InstanceState) shim.InstanceState {
```

Comment:
I'm assuming we are not concerned about changing this technically public interface. I just wanted to call out that we are.

Reply:
Fixed.